### PR TITLE
[2.0.1] Add manifest file parsing (#3105)

### PR DESCRIPTION
**Pull request by sbbroot**
* Add `-m/--manifest` flag to accept manifest.yml produced by
  `epicli init/prepare`

* Add `-v/--verbose` mode for printing out parsed manifest data

* Add ManifestReader class used for paring the manifest.yml file

* Move src/command/*.py to debian/redhat subdirs where needed

### CODE REVIEW

**Comment by seriva** [comment]
@sbbroot Can you resolve the conflicts before review?

**Review by seriva — DISMISSED** [review]

**Comment by seriva** [comment]
@sbbroot Could we pls make a feature branch for this for now and not merge it straight into develop?

**Comment by seriva** [comment]
Also @sbbroot 2 scenarios to think about:

1. People are using it after the prepare step, so they will include their manifest.yml file. Just wondering if in that case there should be a merge/validate procedure in place like we have for apply etc.
2.  People are just including the manifest in apply. do we want apply to also only generate the partial downloading of the requirements from the start?

**Comment by sbbroot** [comment]
@seriva 

1. Yes, there is a separate issue for this: [#3103](https://github.com/epiphany-platform/epiphany/issues/3103)
2. I assume "yes", default manifest.yml alone provides enough information to reduce the number of downloaded files.

File: ansible/playbooks/roles/repository/files/download-requirements/src/config/config.py [hunk 1]
```diff
@@ -93,6 +101,12 @@ def __create_parser(self) -> ArgumentParser:
         parser.add_argument('--no-logfile', action='store_true', dest='no_logfile',
                             help='no logfile will be created')
 
+        parser.add_argument('--verbose', '-v',  action='store_true', dest='verbose',
+                            help='more verbose output will be provided')
+
+        parser.add_argument('--manifest', '-m', metavar='MANIFEST_PATH', type=Path, action='store', dest='manifest',
+                            help='manifest file generated by `epicli apply` and/or `epicli prepare`')
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/src/config/config.py]: Is it possible to use manifest generated by `epicli init --full` or merged version is always required?

File: ansible/playbooks/roles/repository/files/download-requirements/src/config/manifest_reader.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+
+def load_yaml_file(filename: Path) -> Any:
+    return load_yaml_file_all(filename)[0]
+
+
+class ManifestReader:
+    """
+    Load the manifest file and call defined parser methods to process required documents.
+    Main running method is :func:`~manifest_reader.ManifestReader.parse_manifest` which returns formatted manifest output.
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/src/config/manifest_reader.py:26]: Does `~` here has a special meaning or this is typo?

File: ansible/playbooks/roles/repository/tasks/download-requirements.yml [hunk 3]
```diff
@@ -11,6 +11,8 @@
         /var/www/html/epirepo \
         "{{ download_requirements_os_name }}" \
         --repos-backup-file /var/tmp/enabled-system-repos.tar \
+        --verbose \
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/tasks/download-requirements.yml]: How about alphabetical order?

File: ansible/playbooks/roles/repository/files/download-requirements/src/config/manifest_reader.py (hunk 2 above)
Comment (sbbroot) [reply ansible/playbooks/roles/repository/files/download-requirements/src/config/manifest_reader.py:26]: Yes, this has a special meaning in pydoc, it creates a link to that method.

File: ansible/playbooks/roles/repository/files/download-requirements/src/config/config.py (hunk 1 above)
Comment (sbbroot) [reply ansible/playbooks/roles/repository/files/download-requirements/src/config/config.py]: The only minimum data required is: sections `epiphany-cluster` and `configuration/feature-mappings`.
Mappings is implemented in #3119

**Review by to-bar — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Pull request merged**